### PR TITLE
FOIA-368: Bulk report xml uploader.

### PR DESCRIPTION
**Pull request by jesseday**
* Adds a drush command to bulk import a directory of report xml files.
* `drush foia-upload-xml:bulk-upload path/to/report/files` OR `drush foia-upload-xml:bulk-upload /absolute/path/to/report/files`
* includes command alias `drush fuxml path/to/report/files`

### CODE REVIEW

File: docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+   *   migrate_map_foia_agency_report table for the agency and report year
+   *   of the given source file.
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  protected function migrationStatus(FileInterface $file) {
+    $report_data = $this->reportParser->parse($file);
+    $agency = $report_data['agency'] ?? FALSE;
+    $year = $report_data['report_year'] ?? date('Y');
+
+    $status = \Drupal::database()
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php]: Can we use dependency injection instead?

```\Drupal::database()```

becomes

```
use Drupal\Core\Database\Connection;

/**
   * Database connection.
   *
   * @var \Drupal\Core\Database\Connection
   */
  protected $connection;

/**
   * @param \Drupal\Core\Database\Connection $connection
   *     Established database connection.
   */
  public function __construct(Connection $connection) {
    $this->connection = $connection;
  }

$status = $this->connection
->select('migrate_map_foia_agency_report', 'm')
      ->fields('m', ['source_row_status'])
      ->condition('sourceid1', $year)
      ->condition('sourceid2', $agency)
      ->execute()
      ->fetchField();
```

File: docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+      // to be used by the migrations processor.
+      $source = File::create([
+        'uid' => 1,
+        'status' => 0, // temporary
+        'uri' => $filepath,
+      ]);
+
+      try {
+        $this->migrationsProcessor->setSourceFile($source)
+          ->setUser(User::load(1))
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php]: ```User::load(1)```
Can we use dependency injection instead?

File: docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+          ->processAll();
+
+        $status = $this->migrationStatus($source);
+        if ($status === MigrateIdMapInterface::STATUS_FAILED) {
+          throw new \Exception(\Drupal::translation()
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php]: ```\Drupal::translation()```
Can we use dependency injection instead?

File: docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+              '@file' => $filepath,
+            ]));
+        }
+
+        $rows[] = [
+          'file' => $info['basename'],
+          'status' => 'Processed',
+        ];
+      } catch (\Exception $e) {
+        \Drupal::logger('foia_upload_xml')
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php]: ```\Drupal::logger('foia_upload_xml')```
Can we use dependency injection instead?

File: docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+   *
+   * @param string $directory
+   *   The directory path where the xml files live.
+   *
+   * @return array|false
+   *   The xml files contained in the given directory, an empty array if none
+   *   are found, or false on error.
+   */
+  protected function getXmlFiles($directory) {
+    $realpath = FsUtils::realpath($directory);
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php]: ```FsUtils::realpath($directory)```
Can we use dependency injection instead?

File: docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+          'file' => $info['basename'],
+          'status' => 'Skipped, file not readable',
+        ];
+        continue;
+      }
+
+      // This file does not need to be saved or moved, as each file will be
+      // processed in place.  It is wrapped in a FileInterface class in order
+      // to be used by the migrations processor.
+      $source = File::create([
```
Comment (bemarlan) [inline docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php]: ```File::create```
Can we use dependency injection instead?

File: docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php (hunk 5 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php]: Removed the utility usage.

File: docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php (hunk 4 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php]: Done

File: docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php (hunk 3 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php]: Used `dt()`  instead, which seems to be standard in drush command files.  For example, see `docroot/modules/contrib/devel/src/Commands/DevelCommands.php`

File: docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php (hunk 1 above)
Comment (jesseday) [reply docroot/modules/custom/foia_upload_xml/src/Commands/FoiaUploadXmlCommands.php]: done

**Pull request merged**